NEO-TAT/tat_flutter
# Refine simple login result handler for the separated password expiration states

## Description 
In #164 we've tried to use a new flow on login task, however, the school's server seems not allow us to do it.
So after tat_core updated to the latest logic, we tried to apply them into login task in this PR.

## How to Verify? 
Normally login to an account, see what happened. (You may need to logout first)

There are some cases that you may meet:
1. just login..
2. You're username or password not correct -> the login page should remain in front of you.
3. You're account was locked by some reasons -> the login page should remain in front of you.
4. You may need to update your password, otherwise it will expire in xx days -> still can go in to home page but a warning dialog shown.
5. You may need to update your phone number, otherwise it will expire in xx days -> still can go in to home page but a warning dialog shown.
6. Unknown error happened.

## Screenshots/GIF/Test Results (Optional)
